docs/Earn/liquid-staking/matic/matic-liquid-staking-mechanics.md

```diff
@@ -0,0 +1,140 @@
+
+`Polygon Pool` emits a `StakePending(msg.sender, amount)` event.
+
+Ankr Earn Dashboard allows the user to register (import) the received aMATICb tokens to Metamask or another wallet connected to Ethereum.
+
+Ankr backend collects the MATICs staked by users, transferring them from the smart contract to the address controlled by the platform, and makes a validator stake on the Ankr Matic Validator node if the conditions below are met.
+
+Ankr Earn Dashboard can call `aMATICb::balanceOf(address account)` to check the user’s current aMATICb balance.
+
+ANK backend will monitor blockchain events coming from the `PolygonPool` smart contract. Upon a new staking event, the backend:
+
+1. Checks if the newly available MATIC number (*N*) is >= to the minimum threshold. If yes, the available MATICs are sent from the `PolygonPool`  smart contract to the address managed by Ankr. Else, the backend waits for the next stake event from `PolygonPool`. *N* is a config parameter.
+
+2. When MATIC tokens become available at the Ankr address, staking transaction is issued.
+
+3. The Ankr backed monitors staking status and rewards, and restakes the rewards when the sum of the staked MATIC reward is > threshold. The threshold is a config parameter.
+
+
+## Daily exchange ratio updates
+To update the rates, the Ankr backend:
+
+1. Collects information about the total supply of aMATICb.
+
+2. Collects information about MATICs available as stakes on the `MaticPool` smart contract, at the addresses controlled by Ankr, MATICs staked on the Validator node, and MATICs received as reward.
+
+3. Calculates MATIC ratio, diving the MATIC share by the MATICs involved in the staking.
+
+4. Sends a ratio update request to `aMATICb::function updateRatioAndFee(uint256 newRatio, uint256 newFee)`.
+
+
+## Unstaking workflow
+
+The workflow is both user and Ankr-driven. The user part is Step 1, the Ankr part is Steps 2–11.
+
+1. User navigates to the dedicated page in Ankr Earn and unstakes their MATIC.
+
+2. Ankr Earn Dashboard checks the user’s aMATICb balance and shows the user a form to enter the number of MATICs to unstake.
+
+Frontend gets the unstake fee information from Ankr backend:
+
+4. Ankr Earn Dashboard calls Ankr smart contract to approve the transfer fee in Ankr tokens from the user address to the PolygonPool smart contract address. The fee is needed to compensate Ankr for the unstaking expenses.
+
+5. Ankr Earn Dashboard calls `PolygonPool::unstake(uint256 amount, uint256 fee, uint256 useBeforeBlock, bytes memory signature)` to unstake the specified amount of MATICs.
+
+6. `PolygonPool` transfers the equal amount of aMATICb from the user to itself (technically, it locks the amount of aMATICb to unstake in the user’s account).
+
+7. `PolygonPool` issues an unstake event `MaticClaimPending(msg.sender, amount)`.
+
+8. Ankr Earn Dashboard shows a notification to the user, informing about successful unstake request registration.
```

Comment:
displays ....... to indicate that the unstaking request has been registered.

Reply:
thx